src/core/server/services/users/index.ts

```diff
@@ -312,6 +315,92 @@ export async function updatePassword(
   return user;
 }
 
+export async function requestAccountDeletion(
+  mongo: Db,
+  mailer: MailerQueue,
+  tenant: Tenant,
+  user: User,
+  password: string,
+  now: Date
+) {
+  if (!user.email) {
+    throw new EmailNotSetError();
+  }
+
+  const passwordVerified = await verifyUserPassword(user, password, user.email);
+  if (!passwordVerified) {
+    // We throw a PasswordIncorrect error here instead of an
+    // InvalidCredentialsError because the current user is already signed in.
+    throw new PasswordIncorrect();
+  }
+
+  const deletionDate = DateTime.fromJSDate(now).plus({
+    days: SCHEDULED_DELETION_TIMESPAN_DAYS,
+  });
+
+  const updatedUser = await scheduleDeletionDate(
+    mongo,
+    tenant.id!,
+    user.id,
+    deletionDate.toJSDate()
+  );
+
+  const formattedDate = Intl.DateTimeFormat([tenant.locale], {
```

Comment:
Array isn't required:

```suggestion
  const formattedDate = Intl.DateTimeFormat(tenant.locale, {
```